src/components/NavBar/NavigationBar.tsx

```diff
@@ -0,0 +1,208 @@
+        </IconButton>
+    );
+};
+
+const Gap = styled.div`
+    width: 1rem;
+`;
+
+const DrawerButtonWrapper = styled.div`
+    padding: 1rem;
+    border-bottom: 1px solid ${(props) => props.theme.secondaryBackgroundColor};
+
+    &:last-child {
+        border-bottom: none;
+    }
+`;
+
+const DrawerButton = styled(Button)`
+    width: 100%;
+    color: ${(props) => props.theme.secondaryBackgroundColor};
+`;
+
+const ResponsiveAppBar: React.FC<{}> = () => {
+    const [drawer, setDrawer] = React.useState(false);
+
+    const openDrawer = (): void => {
+        setDrawer(true);
+    };
+
+    const closeDrawer = (): void => {
+        setDrawer(false);
+    };
+
+    const pages = [
+        ["Clients", "/clients"],
+        ["Lists", "/lists"],
+        ["Calendar", "/calendar"],
+    ];
+
+    const setThemeMode = useContext(ThemeUpdateContext);
+
+    const switchThemeMode = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        setThemeMode(event.target.checked);
+    };
+
+    return (
+        <>
+            <SwipeableDrawer open={drawer} onClose={closeDrawer} onOpen={openDrawer}>
+                <DrawerInner>
+                    {pages.map(([page, link]) => (
+                        <DrawerButtonWrapper key={page}>
+                            <UnstyledLink href={link} onClick={closeDrawer}>
+                                <DrawerButton variant="text">{page}</DrawerButton>
+                            </UnstyledLink>
+                        </DrawerButtonWrapper>
+                    ))}
+                </DrawerInner>
+            </SwipeableDrawer>
+            <AppBar position="static">
+                <AppBarInner>
+                    <SmallNavBarElement>
+                        <IconButton
+                            size="medium"
+                            aria-label="Mobile Menu Button"
+                            onClick={openDrawer}
+                        >
+                            <MenuIcon />
+                        </IconButton>
+                        <Spacer />
+                    </SmallNavBarElement>
+                    <LogoNavElement />
+                    {/* TODO: VFB-16 need to fix nav colours */}
+                    <LgNavElement>
+                        {pages.map(([page, link]) => (
+                            <UnstyledLink key={page} href={link} onClick={closeDrawer}>
+                                <Button variant="outlined" color="secondary">
+                                    {page}
+                                </Button>
+                            </UnstyledLink>
+                        ))}
+                        <LightDarkSlider onChange={switchThemeMode} />
+                        <Gap />
+                        <SignOutButton />
+                    </LgNavElement>
+                    <SmallNavBarElement>
+                        <Spacer />
+                        <LightDarkSlider onChange={switchThemeMode} key="theme-toggle" />
```

Comment:
why does this have a key but the other one above does not? (can they both get well-named keys?)

Reply:
Added the key of `"light/dark mode switch"` to both